PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/api/data/BidInfo.java

```diff
@@ -54,4 +64,34 @@ public Integer getExp() {
         return exp;
     }
 
+    @Nullable
+    public Map<String, String> getEvents() {
+        return events;
+    }
+
+
+    @NonNull
+    public static BidInfo create(@NonNull ResultCode resultCode, @Nullable BidResponse bidResponse) {
+        if (bidResponse == null) {
+            return new BidInfo(resultCode, null, null);
+        }
+
+        Bid winningBid = bidResponse.getWinningBid();
+        Map<String, String> events = null;
+        if (winningBid != null) {
+            events = winningBid.getEvents();
+        }
+
+        return new BidInfo(resultCode, bidResponse.getTargeting(), events);
+    }
+
+    public static void saveNativeResult(@NonNull BidInfo bidInfo, @Nullable BidResponse bidResponse, @Nullable Object adObject) {
+        if (bidInfo.resultCode == ResultCode.SUCCESS && bidResponse != null) {
+            String cacheId = CacheManager.save(bidResponse.getWinningBidJson());
+            Util.saveCacheId(cacheId, adObject);
+            bidInfo.nativeCacheId = cacheId;
+            bidInfo.exp = bidResponse.getExpirationTimeSeconds();
```

Comment:
Do we set the `exp` value only for native ads? It should be provided to the publisher for any type of bid.